ReactAndroid/src/main/java/com/facebook/react/bridge/JSBundleLoader.java

```diff
@@ -22,18 +22,22 @@
    * should be used. JS bundle will be read from assets directory in native code to save on passing
    * large strings from java to native memory.
    */
-  public static JSBundleLoader createAssetLoader(
-      final AssetManager assetManager,
-      final String assetFileName) {
+  public static JSBundleLoader createFileLoader(
+      final Context context,
+      final String fileName) {
     return new JSBundleLoader() {
       @Override
       public void loadScript(ReactBridge bridge) {
-        bridge.loadScriptFromAssets(assetManager, assetFileName);
+        if (fileName.startsWith("assets://")) {
+          bridge.loadScriptFromAssets(context.getAssets(), fileName.replaceFirst("assets://", ""));
+        } else {
+          bridge.loadScriptFromFile(fileName, fileName);
+        }
       }
 
       @Override
       public String getSourceUrl() {
-        return "file:///android_asset/" + assetFileName;
+        return fileName;
```

Comment:
I think this is only used to provide source URL to JSC so that it will display stack traces correctly (JSC won't try to open that file, it just require to have that name, otherwise stack traces are empty). I don't think we attempt to read this anywhere else (otherwise `assets://`schema won't work) but not sure about that
cc @foghina